### PR TITLE
refactor(KP-330): Typescript setup

### PR DESCRIPTION
**Pull request by axelbjo**

### CODE REVIEW

**Review by amirhossein-haerian — APPROVED** [review]
LGTM 🚀

**Comment by belanglos** [comment]
Great job! Really happy about this 🚀 😃

**Review by karlandindrakryggen — CHANGES_REQUESTED** [review]
I think you should fix the problem with `noImplicitAny`, probably by adding `allowJs`.

I guess you normally could skip `ourDir` when using it with webpack, but I got an warning about it when using `allowJs` because the default is to ourput in same dir (that would cause js files to be overwritten by its own output). So I would say that we can keep outDir even if its not used.

File: webpack.config.js [hunk 1]
```diff
@@ -87,6 +85,13 @@ function getTransformationRules({ contextIsNode, subDir = null }) {
             options: { ...BabelConfig, ...ALLOW_MIX_OF_CJS_AND_ESM_IMPORTS },
           },
         },
+        {
+          test: /\.tsx?$/,
```
[the remark this answers is not in the document]
Comment (axelbjo) [reply webpack.config.js]: We have updated it so it handles .ts and .tsx files now

File: public/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "compilerOptions": {
+    "outDir": "./dist/",
+    "noImplicitAny": true,
```
[the remark this answers is not in the document]
Comment (axelbjo) [reply public/tsconfig.json:4]: "allowJs": true and "checkJs": false have been added to tsconfig.json now

**Review by karlandindrakryggen — APPROVED** [review]

**Pull request merged**